status/reporter_gitlab.go

```diff
@@ -56,103 +57,105 @@ func NewGitLabReporter(logger logr.Logger, k8sClient client.Client) *GitLabRepor
 var _ ReporterInterface = (*GitLabReporter)(nil)
 var existingCommitStatus *gitlab.CommitStatus
 
-// Detect if snapshot has been created from gitlab provider
-func (r *GitLabReporter) Detect(snapshot *applicationapiv1alpha1.Snapshot) bool {
-	return metadata.HasAnnotationWithValue(snapshot, gitops.PipelineAsCodeGitProviderLabel, gitops.PipelineAsCodeGitLabProviderType) ||
-		metadata.HasLabelWithValue(snapshot, gitops.PipelineAsCodeGitProviderAnnotation, gitops.PipelineAsCodeGitLabProviderType)
+// Detect if object has label/annotation for gitlab provider
+func (r *GitLabReporter) Detect(obj metav1.Object) bool {
+	return metadata.HasAnnotationWithValue(obj, gitops.PipelineAsCodeGitProviderLabel, gitops.PipelineAsCodeGitLabProviderType) ||
+		metadata.HasLabelWithValue(obj, gitops.PipelineAsCodeGitProviderAnnotation, gitops.PipelineAsCodeGitLabProviderType) ||
+		metadata.HasAnnotationWithValue(obj, tekton.PipelineAsCodeGitProviderLabel, gitops.PipelineAsCodeGitLabProviderType) ||
+		metadata.HasLabelWithValue(obj, tekton.PipelineAsCodeGitProviderAnnotation, gitops.PipelineAsCodeGitLabProviderType)
 }
 
 // GetReporterName returns the reporter name
 func (r *GitLabReporter) GetReporterName() string {
 	return "GitlabReporter"
 }
 
-// Initialize initializes gitlab reporter
-func (r *GitLabReporter) Initialize(ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot) error {
+// Initialize initializes gitlab reporter for snapshot or build pipelinerun
+func (r *GitLabReporter) Initialize(ctx context.Context, obj metav1.Object) error {
 	var unRecoverableError error
-	token, err := GetPACGitProviderToken(ctx, r.k8sClient, snapshot)
+	token, err := GetPACGitProviderToken(ctx, r.k8sClient, obj)
 	if err != nil {
-		r.logger.Error(err, "failed to get PAC token from snapshot",
-			"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
+		r.logger.Error(err, "failed to get PAC token from object",
```

Comment:
@dirgim Sorry for the later response. I meant to get the kind of object in log or error message. But the `Object` in "k8s.io/apimachinery/pkg/apis/meta/v1" doesn't support GetKind() function. Thanks!

Reply:
I did some search and test and foundd the TypeMeta is lost when transforming data, and I plan to define a function to get object type as below. Thank you.
```
func GetObjectType(obj metav1.Object) string {
	if _, ok := obj.(*applicationapiv1alpha1.Snapshot); ok {
		return "Snapshot"
	}
	if _, ok := obj.(*tektonv1.PipelineRun); ok {
		return "BuildPipelinerun"
	}
	return ""
}
```